packages/phase2cli/src/commands/setup.ts

```diff
@@ -58,15 +68,418 @@ import {
+ * @param zkeyLocalPathAndFileName <string> - the local complete path of the pre-computed zKey selected file.
+ * @returns <Promise<boolean>> - the validity of the pre-computed zKey.
+ */
+const handlePreComputedZkeyVerification = async (
+    r1csLocalPathAndFileName: string,
+    potLocalPathAndFileName: string,
+    zkeyLocalPathAndFileName: string
+): Promise<boolean> => {
+    console.log(
+        `${theme.symbols.info} Checking the pre-computed zKey locally on your machine (to avoid any R1CS, PoT, zKey combination errors)`
+    )
+
+    // Verify validity of pre-computed zKey (R1CS and PoT are implicitly verified).
+    const valid = await zKey.verifyFromR1cs(
+        r1csLocalPathAndFileName,
+        potLocalPathAndFileName,
+        zkeyLocalPathAndFileName,
+        console
+    )
+
+    await sleep(3000) // workaround for unexpected file descriptor close.
+
+    if (valid) console.log(`${theme.symbols.success} The provided pre-computed zKey has passed validation check`)
+    else {
+        console.log(`${theme.symbols.error} The provided pre-computed zKey is invalid!`)
+
+        // Prompt to generate a new zKey from scratch.
+        const newZkeyGeneration = await promptZkeyGeneration()
+
+        if (!newZkeyGeneration) showError(COMMAND_ERRORS.COMMAND_SETUP_ABORT, true)
+    }
+
+    return valid
+}
+
+/**
+ * Generate a brand new zKey from scratch.
+ * @param r1csLocalPathAndFileName <string> - the local complete path of the R1CS selected file.
+ * @param potLocalPathAndFileName <string> - the local complete path of the PoT selected file.
+ * @param zkeyLocalPathAndFileName <string> - the local complete path of the pre-computed zKey selected file.
+ */
+const handleNewZkeyGeneration = async (
+    r1csLocalPathAndFileName: string,
+    potLocalPathAndFileName: string,
+    zkeyLocalPathAndFileName: string
+) => {
+    console.log(
+        `${theme.symbols.info} The computation of your brand new zKey is starting soon.\n${theme.text.bold(
+            `${theme.symbols.warning} Be careful, stopping the process will result in the loss of all progress achieved so far.`
+        )}`
+    )
+
+    // Generate zKey.
+    await zKey.newZKey(r1csLocalPathAndFileName, potLocalPathAndFileName, zkeyLocalPathAndFileName, console)
+
+    console.log(`\n${theme.symbols.success} Generation of genesis zKey completed successfully`)
+}
+
+/**
+ * Manage the creation of a ceremony file storage bucket.
+ * @param firebaseFunctions <Functions> - the Firebase Cloud Functions instance connected to the current application.
+ * @param ceremonyPrefix <string> - the prefix of the ceremony.
+ * @returns <Promise<string>> - the ceremony bucket name.
+ */
+const handleCeremonyBucketCreation = async (firebaseFunctions: Functions, ceremonyPrefix: string): Promise<string> => {
+    // Compose bucket name using the ceremony prefix.
+    const bucketName = getBucketName(ceremonyPrefix, process.env.CONFIG_CEREMONY_BUCKET_POSTFIX!)
```

Comment:
We are never checking the existence of this .env var

Reply:
Same as above, check L40-41 in `lib/services.ts`